k8s/examples/fencing/README.md

```diff
@@ -0,0 +1,114 @@
+# Fencing StatefulSets on Kubernetes with StorageOS
+
+This example shows an example of how to lower the time to recover for a MySQL
+StatefulSet on Kubernetes with StorageOS fencing, in the case of node failure.
+Normally StatefulSet pods are not rescheduled by Kubernetes when nodes become
+unavailable due to the guarantee of StatefulSet pod uniqueness.
+
+For more information regarding fencing please see the [StorageOS fencing
+documentation](https://docs.storageos.com/docs/concepts/fencing).
+
+## Deploy
+
+In order to deploy MySQL you just need to clone this repository and use
+kubectl to create the Kubernetes objects.
+
+```bash
+$ git clone https://github.com/storageos/deploy.git storageos
+$ cd storageos
+$ kubectl create -f ./k8s/examples/fencing
+```
+If you inspect the StatefulSet manifest you'll see that two StorageOS labels
+are applied to the pod. `storageos.com/replicas` and `storageos.com/fenced`
+labels are required for fencing to work. In order to fence a pod it must; mount
+at least one StorageOS volume, all StorageOS volumes mounted by the pod must
+have a healthy replica and `DISABLE_FENCING` is not set.
+
+Once this is done you can check that a mysql pod is running
+```bash
+$ kubectl get pods -w -l app=mysql
+   NAME        READY    STATUS    RESTARTS    AGE
+   client      1/1      Running    0          1m
+   mysql-0     1/1      Running    0          1m
+```
+
+Connect to the MySQL client pod and connect to the MySQL server through the
+service.
+```bash
+$ kubectl exec -it client -- mysql -h mysql-0.mysql -u root
+mysql> create database shop;
+mysql> use shop;
+mysql> create table books (title VARCHAR(256), price decimal(4,2));
+mysql> insert into books value ('Gates of Fire', 13.99);
+mysql> select * from books
++---------------+-------+
+| title         | price |
++---------------+-------+
+| Gates of Fire | 13.99 |
++---------------+-------+
+1 row in set (0.00 sec)
+```
+
+## Node Failure
+
+Check what node the mysql-0 pod is running on and make that node unavailable
+e.g. shutdown the node or stop the kubelet on the node. Now watch as the
+mysql-0 pod is rescheduled onto a different node.
+
+Note that if you are using the CSI driver there is a CSI helper pod that can be
+running as either a StatefulSet or as a Deployment. Given the nature of
+StatefulSets make sure that the mysql-0 pod and the storageos-statefulset-0 pod
```

Comment:
I'd recommend the use of deployments by default rather than saying that StorageOS has fix it. It's good to mention about the statefulset to ensure they are not using that mode, or even be careful with it. But not to say that there is a problem and we fix it...

I merged the PR to ensure all the CR examples use the deployment rather than statefulset